sycl/source/device_selector.cpp

```diff
@@ -28,11 +29,60 @@ static bool isDeviceOfPreferredSyclBe(const device &Device) {
          backend::opencl;
 }
 
+// @return True if the device is invalid for the current backend preferences
+static bool isDeviceInvalidForBe(const device &Device) {
+
+  if (Device.is_host())
+    return false;
+
+  // Taking the version information from the platform gives us more useful
+  // information than the driver_version of the device.
+  const platform platform = Device.get_info<info::device::platform>();
+  const std::string platformVersion =
+      platform.get_info<info::platform::version>();
+
+  backend *BackendPref = detail::SYCLConfig<detail::SYCL_BE>::get();
+  auto BackendType = detail::getSyclObjImpl(Device)->getPlugin().getBackend();
+  static_assert(std::is_same<backend, decltype(BackendType)>(),
+                "Type is not the same");
+
+  // If no preference, assume OpenCL and reject CUDA backend
+  if (BackendType == backend::cuda && !BackendPref) {
```

Comment:
this always disqualifies cuda devices when `SYCL_BE` is not set, right...?

Reply:
Yes, OpenCL is the preferred path...